### ISSUE TITLE
OpenSSF CII Best Practices badge

### ISSUE THREAD

**Issue by ljharb** [fastify/fastify#4917]
Hi folks! I chatted with @mcollina and he suggested I file an issue.

For my @OpenJS-foundation "Security Engineering Champion" role, i'm working on getting the [OpenSSF best practices badge](https://bestpractices.coreinfrastructure.org/) on a bunch of OpenJS projects. I did [nvm](https://github.com/nvm-sh/nvm), and now i'm planning to work on fastify. I'm hoping that's OK with all of you?

I've created a [best practices project](https://bestpractices.coreinfrastructure.org/en/projects/7585) myself, and can later add any maintainers who create an account and give me their user ID (so yall can maintain it moving forward).

Assuming this is OK, I'll likely be asking questions as i go to clarify, but the intention is to spare you folks the majority of the work to document what you're already doing.

**Comment by Uzlopak** [comment]
We are talking about only fastify-core, right? I guess, we can get alot of the points green. 

When I see the gold(?) standard, than it shows e.g. that fastify-core is not setting nosniff. But this is covered by @fastify/helmet. Just because we dont send it in core doesnt mean that we dont support it. Do we have to put some more documentation and reference to @fastify/helmet to meet such criterias? 

Issue could be e.g. that you require to have a changelog/release notes file, despite that we use the github releases functionality, where the changes can be seen. Or is that also enough?

But some gold(?) stuff we will probably not meet, e.g. Copyright notice in every file. 

How do we meet e.g. criterias like that fastify uses semver? Do we need to create a VERSIONING.md were we describe that we use semver or is e.g. a github badge like this enough?

[![versioning](https://img.shields.io/badge/versioning-standard-brightgreen.svg?style=flat)](https://semver.org/)

**Comment by jsumners** [comment]
What does this net us aside from more maintenance?

**Comment by ljharb** [comment]
@Uzlopak the goal isn't necessarily to hit gold; passing is the intention, and silver or gold is a nice-to-have. Personally I suspect that using Github releases satisfies the changelog requirement - but looking into that sort of thing is exactly the work I'm hoping to do on yall's behalf.

As for a copyright notice in every file, that's an archaic and harmful practice that I'm already pushing back on with OpenSSF, so rest assured I wouldn't suggest following that one :-) For semver, I think that for the npm ecosystem, it's _assumed_ everyone is using semver, such that if they're not they'd need to call it out.

@jsumners ideally, you're already following all the best practices, and there's no work to be done. Potentially, the security of the project could be improved. Additionally, all OpenJS Foundation projects will eventually be expected to at least achieve "passing" status of this badge, so this gives fastify a leg up.

**Comment by Uzlopak** [comment]
I think we can do a little bit better on some static analysis tools. Like using eslint-plugin-sonar or activating codeql, which we should investigate.

In the gold section:
I think we meet the 2 unassociated active contributers, because nobody gets paid etc ..

![Screenshot_20230719-084304_Chrome](https://github.com/fastify/fastify/assets/5059100/822d0ccb-0a3e-4b78-ab7c-1a1b6d0def58)
![Screenshot_20230719-084310_Chrome](https://github.com/fastify/fastify/assets/5059100/ebad5669-04f4-4688-8f3f-454299911b23)
![Screenshot_20230719-084418_Chrome](https://github.com/fastify/fastify/assets/5059100/dc7c1c46-80f9-415b-b00a-71f33c0d3367)

Regarding marking small tasks for interested devs to contribute, we mark github issues with the 'good-first-issue' tag and publish them on the fastify website

https://fastify.dev/contribute/

**Comment by metcoder95** [comment]
>@jsumners ideally, you're already following all the best practices, and there's no work to be done. Potentially, the security of the project could be improved. Additionally, all OpenJS Foundation projects will eventually be expected to at least achieve "passing" status of this badge, so this gives fastify a leg up.

I agree with this statement of @ljharb in the terms that it seems we are mostly doing right, but smaller pieces are required to achieve the `passing` badge. I believe that it will require some coordination on our side to achieve the next step (e.g. code analysis tool [although I'm not a fan of them 😅], testing policies, etc.).

We can start small and open discussions about what's required for moving forward and assess what and how we can do it (and maybe even if want to do it 😅)

**Comment by Uzlopak** [comment]
Well, it seems that codeql is somehow matured, because i think github wants to charge my employer additionally to the normal user licenses more than 500€ per user if we activate it (we have 1000 users, a little bit high price for something devs anyway ignore lol)
Also i think codeql makes more sense if we actually learn how to do the ql part.

**Comment by Fdawgs** [comment]
> code analysis tool [although I'm not a fan of them 😅]

I'm happy to set up CodeQL in GitHub Actions and do the grunt work of flushing out the false positives for this btw gang.

**Comment by ljharb** [comment]
It sounds like everyone’s on board. The first step is for me to go through the auditing process to figure out where there’s gaps :-) we can worry about making improvements after those are identified. I’m happy to see you all so enthusiastic tho!

**Comment by Uzlopak** [comment]
@ljharb 

Can add me please?
https://bestpractices.coreinfrastructure.org/en/users/28764

**Comment by ljharb** [comment]
Done!

**Comment by Fdawgs** [comment]
>     * > The project MUST give credit to the reporter(s) of all vulnerability reports resolved in the last 12 months, except for the reporter(s) who request anonymity. If there have been no vulnerabilities resolved in the last 12 months, select "not applicable"

Does https://github.com/fastify/fastify/security/advisories cover this?

**Comment by Uzlopak** [comment]
> The project MUST give credit to the reporter(s) of all vulnerability reports resolved in the last 12 months, except for the reporter(s) who request anonymity. If there have been no vulnerabilities resolved in the last 12 months, select "not applicable"

This is happening. See https://github.com/fastify/fastify/security/advisories 
Also if CVE are issued, we can have credits with real name afaik

> The project SHOULD support multiple cryptographic algorithms, so users can quickly switch if one is broken. Common symmetric key algorithms include AES, Twofish, and Serpent. Common cryptographic hash algorithm alternatives include SHA-2 (including SHA-224, SHA-256, SHA-384 AND SHA-512) and SHA-3.

Not applicable here, but we write our plugins usually configurable and try to be unopinionated, e.g. https://github.com/fastify/fastify-cookie/issues/216#issuecomment-1275229780

> The project MUST support storing authentication credentials (such as passwords and dynamic tokens) and private cryptographic keys in files that are separate from other information (such as configuration files, databases, and logs), and permit users to update and replace them without code recompilation. If the project never processes authentication credentials and private cryptographic keys, select "not applicable" (N/A).

We have no authentication here, so N/A. Our plugins are unoninionated regarding the implementation. So it is the implementers duty to take care of that. 

> The default security mechanisms within the software produced by the project MUST NOT depend on cryptographic algorithms or modes with known serious weaknesses (e.g., the SHA-1 cryptographic hash algorithm or the CBC mode in SSH).

We arent dependent on any cryptographic algorithm in fastify-core

> The software produced by the project SHOULD support secure protocols for all of its network communications, such as SSHv2 or later, TLS1.2 or later (HTTPS), IPsec, SFTP, and SNMPv3. Insecure protocols such as FTP, HTTP, telnet, SSLv3 or earlier, and SSHv1 SHOULD be disabled by default, and only enabled if the user specifically configures it. If the software produced by the project does not support network communications, select "not applicable"

We support what nodejs supports. But imho by default we use http.

**Comment by ljharb** [comment]
Awesome, thanks!

Looks like the only items left to get a "passing" badge are related to dynamic analysis - something like CodeQL would indeed suffice, so thanks to @Fdawgs for volunteering to do that. https://github.com/fastify/fastify/issues/4917#issuecomment-1641640942

For "silver", the remaining items (that aren't blocked on "passing", adding the badge, or an OpenSSF response) are:
 - having a DCO/CLA, a requirement i'm not personally a fan of
 - having a documented roadmap (does this exist?)
 - not defaulting to http
 - cryptographically signed releases and tags
 - > The project results MUST check all inputs from potentially untrusted sources to ensure they are valid (an *allowlist*), and reject invalid inputs, if there are any restrictions on the data at all.

**Comment by Uzlopak** [comment]
documented roadmap is kind of strange for our needs. We are actually most of the times fixing bugs, adapting feature requests or improving the performance. We orientate us on the nodejs cyclus and use EOL of nodejs versions to bump our major, where we also focus on our breaking changes. Lets be honest: Fastify is more like a low overhead wrapper around nodejs. I have not the feeling that we a precise idea what we want to implement in a year. Its a framework and not a product, where you constantly add features based on a bigger plan or so. Hard to describe. 

defaulting to https:
That would be kind of annoying and not very nice for the DX. How would you differentiate between a local development server and a production server? I would say, that we actually expect that fastifyjs is behind a reverse proxy/load balancer like traefik, which takes care of the https part. Maybe others can give a statement regarding this. Maybe we need more documentation, that clarifies this?

Is this considered as cryptographically signed releases and tags?
![Screenshot from 2023-07-20 00-33-33](https://github.com/fastify/fastify/assets/5059100/ce4d64c8-eb85-47e3-b6e0-600ee9c7bfd8)

> The project results MUST check all inputs from potentially untrusted sources to ensure they are valid (an allowlist), and reject invalid inputs, if there are any restrictions on the data at all.

We are not providing a product, but a framework. Only the implementer knows, what valid or invalid inputs are. We are providing the appropriate configuration options to the input validation.

**Comment by climba03003** [comment]
Can you add me as well?
https://bestpractices.coreinfrastructure.org/en/users/28772

> documented roadmap is kind of strange for our needs.

> having a documented roadmap (does this exist?)

Documented one (actual file) would be too heavy for us. For the upcoming release, we use `tag` to mark the must DO item.
For example [`v5`](https://github.com/fastify/fastify/issues?q=is%3Aopen+is%3Aissue+label%3Av5.x) which is the upcoming release.
Is that sufficient, or is GitHub Project Board can classify as documented?

> Is this considered as cryptographically signed releases and tags?

I believe yes, but not all signed.

**Comment by ljharb** [comment]
@climba03003 done! and thanks, i think that can count as a roadmap.

How would you consider Github releases cryptographically signed? Do all of you use GPG keys when making tags?

**Comment by climba03003** [comment]
> How would you consider Github releases cryptographically signed?

For the core, it mostly @mcollina doing the job and yes he is using GPG keys.
For the plugin, it mostly Matteo, me and @Eomm. I am also using GPG keys starting from this year.

GitHub doing the great job for checking if the release or tag is signed by providing the green mark.
For example
![image](https://github.com/fastify/fastify/assets/23028015/02c47349-97e8-4d7f-8838-1827ba051275)

**Comment by Fdawgs** [comment]
> > How would you consider Github releases cryptographically signed?
> 
> For the core, it mostly @mcollina doing the job and yes he is using GPG keys. For the plugin, it mostly Matteo, me and @Eomm. I am also using GPG keys starting from this year.
> 
> GitHub doing the great job for checking if the release or tag is signed by providing the green mark. For example ![image](https://user-images.githubusercontent.com/23028015/254784526-02c47349-97e8-4d7f-8838-1827ba051275.png)

Whilst we're on the topic, we could enforce requiring signed commits for the `main` branches across the repos?
At present we don't do it, see below for the main fastify repo:

![image](https://github.com/fastify/fastify/assets/43814140/1bef130f-a074-41f7-bd2c-4bbf1913a477)

Seems to be on a branch-by-branch setting though, can't do it repo/org wide. :|

**Comment by Uzlopak** [comment]
I wonder, if this works properly.

When i do changes using github and i guess codespaces than my commits are digitally signed and verified. If i do changes on my local machine it nevers shows the digitally signed badge in my commits, despite using ssh to clone and work with the repos.
I also dont know how it fits with our squash merges.

**Comment by climba03003** [comment]
> When i do changes using github and i guess codespaces than my commits are digitally signed and verified.

Yes, GitHub always using `git` with `gpg` when using the web editor.

> If i do changes on my local machine it nevers shows the digitally signed badge in my commits, despite using ssh to clone and work with the repos.

You need to properly uploads your GPG keys to GitHub or download the GPG keys generated by GitHub.
More importantly, configures `git` to use `gpg` before each commit. 
If you do not configures `git` to use `gpg`, it will never sign any commit.

**Comment by Uzlopak** [comment]
Probably the last one has to be done, because for my employer i had to set up gpg anyway.

**Comment by Fdawgs** [comment]
> I wonder, if this works properly.
> 
> When i do changes using github and i guess codespaces than my commits are digitally signed and verified. If i do changes on my local machine it nevers shows the digitally signed badge in my commits, despite using ssh to clone and work with the repos. I also dont know how it fits with our squash merges.

GitHub sign [web-based](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification#statuses-with-vigilant-mode-enabled) and [codespace](https://docs.github.com/en/codespaces/managing-your-codespaces/managing-gpg-verification-for-github-codespaces) (if enabled) ones on your behalf.

And to show the signed badge, you need do what @climba03003 said, as well as potentially [enabling vigilant mode in your account settings](https://docs.github.com/en/authentication/managing-commit-signature-verification/displaying-verification-statuses-for-all-of-your-commits).

**Comment by Uzlopak** [comment]
Will check in 30 min. But what does it mean for devs who want go contribute? Do we they are also forced to setup gpg with git?

**Comment by climba03003** [comment]
> But what does it mean for devs who want go contribute?

Squash and Merge will be handled by GitHub. All the commits after squash will be signed using the GitHub GPG keys.
One exceptional case is flag with [vigilant mode](https://docs.github.com/en/authentication/managing-commit-signature-verification/displaying-verification-statuses-for-all-of-your-commits) may display `Partial Verified`.

Anyway, it means the git history should be either `Verified` or `Partial Verified` only. Unless someone push directly without sign.

> Do we they are also forced to setup gpg with git?

No

**Comment by metcoder95** [comment]
Can I also be added? https://bestpractices.coreinfrastructure.org/en/users/28752

>Anyway, it means the git history should be either Verified or Partial Verified only. Unless someone push directly without signing.

Should we ensure that all core members with the ability to do release should have `git` + `gpg` to sign the tags?
I think that's not verifiable on our end, but mostly due work each of us will have to do.

>not defaulting to http

Is there anything that can be done around this, I'm not sure if this is something we should pursue. As was said earlier in the thread, most of the time Fastify is used behind a reverse proxy, or under an architecture where the TLS is handled at a different end and not Fastify itself.

>Documented one (actual file) would be too heavy for us. For the upcoming release, we use tag to mark the must DO item.
For example [v5](https://github.com/fastify/fastify/issues?q=is%3Aopen+is%3Aissue+label%3Av5.x+sort%3Aupdated-desc) which is the upcoming release.
Is that sufficient, or is GitHub Project Board can classify as documented?

I imagine we will require to link the PRs/Issues WIP and already merged to be reflected there?

**Comment by jsumners** [comment]
I have been keeping up with the responses on this thread. I have to say that I am against this initiative. The short summary is that this initiative seems to clearly be designed in the same vein as "enterprise" security policies and ignores that it is being applied to open source projects.

> Additionally, all OpenJS Foundation projects will eventually be expected to at least achieve "passing" status of this badge, so this gives fastify a leg up.

@mcollina I retract my "sure, go ahead" vote in regard to Pino.

> * having a DCO/CLA, a requirement i'm not personally a fan of

I am also not a fan and I am 100% against this one. We _want_ contributions from the community and these things just make people say "nah, I have better things to do with my time." To be clear: I am one of those people and _will not_ bother with a project that requires one of these unless I am being paid to do so.

> * having a documented roadmap (does this exist?)

We don't have a roadmap and I do not see that changing. This project works very much like the core Node.js project: people contribute the things they want to see in the framework. The core maintainers are doing primarily that: maintaining what exists.

> * not defaulting to http

I am strongly against changing this. As @Uzlopak stated, we [highly recommend people use a reverse proxy](https://fastify.dev/docs/latest/Guides/Recommendations#use-a-reverse-proxy) to front their server and terminate TLS at the proxy. Defaulting to HTTPS would make the developer experience much more miserable and good DX is one of our goals. Note, we _do_ promote secure defaults. See https://fastify.dev/docs/latest/Reference/Server#listen as an example (we don't listen on `0.0.0.0`).

> * cryptographically signed releases and tags

This one I agree with. It's _super easy_ now that GitHub support SSH key signing. As others have pointed out, this is already done by the fact that @mcollina is issuing releases and he signs his commits. I _do not_ think we need to manually build and sign tarballs. 

> * The project results MUST check all inputs from potentially untrusted sources to ensure they are valid (an _allowlist_), and reject invalid inputs, if there are any restrictions on the data at all.

I do not understand how this relates to this project. This is one of those blanket requirements that ignores what it is being applied to, as best I can tell.

-----

> Whilst we're on the topic, we could enforce requiring signed commits for the `main` branches across the repos?

I am quite sure that would make contributions to the project prohibitively difficult. Anyone who wants to submit a PR will need to enable commit signing:

1. https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification#signature-verification-for-rebase-and-merge
2. https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-protected-branches/about-protected-branches#require-signed-commits

**Comment by mcollina** [comment]
> I have been keeping up with the responses on this thread. I have to say that I am against this initiative. The short summary is that this initiative seems to clearly be designed in the same vein as "enterprise" security policies and ignores that it is being applied to open source projects.

What's the main issue for you? Apart a few quirks, this seems mostly innocuous and we are already passing the vast majority of the requirements.
Ideally we should be able to get the badge with minimal work because we are following all the guidelines that are relevant to us already.

> > Additionally, all OpenJS Foundation projects will eventually be expected to at least achieve "passing" status of this badge, so this gives fastify a leg up.

I don't see this happening anytime soon tbh. I would rather keep that out of the discussion.

> @mcollina I retract my "sure, go ahead" vote in regard to Pino.

Ok.

> > * having a DCO/CLA, a requirement i'm not personally a fan of
> 
> 
> 
> I am also not a fan and I am 100% against this one. We _want_ contributions from the community and these things just make people say "nah, I have better things to do with my time." To be clear: I am one of those people and _will not_ bother with a project that requires one of these unless I am being paid to do so.

We already have this requirement due to being a project member of OpenJS. Neither Node.js or us are actually enforcing it in any way atm. We have a checkbox in the PR template right now.

> > * having a documented roadmap (does this exist?)
> 
> 
> 
> We don't have a roadmap and I do not see that changing. This project works very much like the core Node.js project: people contribute the things they want to see in the framework. The core maintainers are doing primarily that: maintaining what exists.

Same.

> > * not defaulting to http
> 
> 
> 
> I am strongly against changing this. As @Uzlopak stated, we [highly recommend people use a reverse proxy](https://fastify.dev/docs/latest/Guides/Recommendations#use-a-reverse-proxy) to front their server and terminate TLS at the proxy. Defaulting to HTTPS would make the developer experience much more miserable and good DX is one of our goals. Note, we _do_ promote secure defaults. See https://fastify.dev/docs/latest/Reference/Server#listen as an example (we don't listen on `0.0.0.0`).

Me too. My 2 cent on this one is that it is a N/A type of answer.

> > * cryptographically signed releases and tags
> 
> 
> 
> This one I agree with. It's _super easy_ now that GitHub support SSH key signing. As others have pointed out, this is already done by the fact that @mcollina is issuing releases and he signs his commits. I _do not_ think we need to manually build and sign tarballs. 

This is something we should be doing anyway... but it requires a bit of automation work for all releases. After we do that.. it's kind of a breeze.

> 
> 
> > Whilst we're on the topic, we could enforce requiring signed commits for the `main` branches across the repos?


Hard no. Signed commits creates only issue for newcomers and are at odds with making the project easy to contribute with.

**Comment by jsumners** [comment]
> What's the main issue for you? Apart a few quirks, this seems mostly innocuous and we are already passing the vast majority of the requirements.

If we do not have to change anything, that is fine. My issue is that a policy has been created that we must adhere to in which the policy does not consider the project(s) it is being applied to. That's where those quirks come from, and we see them listed throughout this thread.

> We already have this requirement due to being a project member of OpenJS. Neither Node.js or us are actually enforcing it in any way atm. We have a checkbox in the PR template right now.

If you are referring to https://github.com/fastify/fastify/blob/bf01ba4438fd662361a23087d23056caa044aa5e/CONTRIBUTING.md#developers-certificate-of-origin-11, then I am okay with that. What I am against is requiring hoops to jump through like https://github.com/open-telemetry/community/blob/24d0472a852699f2dcd3766525568e2a66dcea54/CONTRIBUTING.md#sign-the-cla.

**Comment by mcollina** [comment]
> If we do not have to change anything, that is fine. My issue is that a policy has been created that we must adhere to in which the policy does not consider the project(s) it is being applied to. That's where those quirks come from, and we see them listed throughout this thread.

There are many different projects, even ones with complex infrastructure behind them (for example, Node.js has one of the largest Jenkins CI in existence). The policy _should_ be dev-friendly, and if it's not @ljharb can definitely go back to them and improve things. The OpenSSF crowd is usually very receptive to maintainers requirements.

I actually reviewed initial work on that policy long time ago and things mostly make sense to me.

> If you are referring to https://github.com/fastify/fastify/blob/bf01ba4438fd662361a23087d23056caa044aa5e/CONTRIBUTING.md#developers-certificate-of-origin-11, then I am okay with that. What I am against is requiring hoops to jump through like https://github.com/open-telemetry/community/blob/24d0472a852699f2dcd3766525568e2a66dcea54/CONTRIBUTING.md#sign-the-cla.

Agreed.

----

@ljharb according to your list of missing things:

> * having a DCO/CLA, a requirement i'm not personally a fan of

https://github.com/fastify/fastify/blob/bf01ba4438fd662361a23087d23056caa044aa5e/CONTRIBUTING.md#developers-certificate-of-origin-11

(similar to Node.js this is not enforced in any way)

> * having a documented roadmap (does this exist?)

Saying that we plan a major release every year or two is good enough? 
We typically use GitHub Milestones to track the work on major releases. Here is the one for v5: https://github.com/fastify/fastify/issues?q=is%3Aopen+is%3Aissue+milestone%3Av5.0.0.

I _guess_ this count as a documented roadmap.

> * not defaulting to http

I understand this is N/A for us, in the same way Node.js provides `http.createServer()` and `https.createServer()`. It's up to the users. All our websites and infra is HTTPs only.

> * cryptographically signed releases and tags

This is missing. It's mostly due to missing work to make it happen. I have a full set up demo for this at https://github.com/mcollina/test-releases/blob/main/.github/workflows/publish-releases.yml and https://github.com/mcollina/test-publish/blob/main/.github/workflows/publish.yml. By combining these two we should have automated signed releases on npm.

> * The project results MUST check all inputs from potentially untrusted sources to ensure they are valid (an allowlist), and reject invalid inputs if there are any restrictions on the data at all.

This is implemented. We provide a safe mechanism for users to do so, but it's up to them. We even provide protection against prototype poisoning by default.

**Comment by Uzlopak** [comment]
I thought about the http by default.

Fastify by itself is not a server. You can create servers with it. 

So maybe it if we use the definition that product are the implementations with fastify, then we are at:

> The software produced by the project MUST use, by default, only cryptographic protocols and algorithms that are publicly published and reviewed by experts (if cryptographic protocols and algorithms are used). [crypto_published]

So we are back to step 1:
Create a https server by default. 

And we can imho not detect if we are in production environment. We habe no NODE_ENV.
I dont see any way to get it conform and really safe. I mean... We could say by access from localhost/127.0.0.1 we can have http connection and any other remote ip address we must provide pem files and stuff or have trustProxy set to true. Or maybe add an Option called trustAddress or so, where we specify ip addresses from where we can have http requests. 
But i assume that this comes with perf penalties.

**Comment by ljharb** [comment]
@metcoder95 added!

Re http, I'm convinced, it's N/A and you're fine :-)

I'll update the answers based on these responses and come back.

@jsumners you are quite right that the best practices questionnaire is designed to cover a wide and disparate variety of projects, and thus many of the things may not apply or may not be worded precisely. These are _guidelines_ though, and as @mcollina says, OpenSSF will absolutely respond to our feedback, and there's a lot of subjectivity applied anyways - so as long as an answer is defensible, it'll be fine.

I agree that project maintainers should never feel _forced_ to make changes as a result of this initiative - only changes they all agree are improvements.

**Comment by mcollina** [comment]
> For what it's worth, I was referring to signed tarballs on the GitHub releases pages.

I don't think signed tarballs would help anybody, the npm signed releases process is actually quite good.

**Comment by Fdawgs** [comment]
> > code analysis tool [although I'm not a fan of them 😅]
> 
> I'm happy to set up CodeQL in GitHub Actions and do the grunt work of flushing out the false positives for this btw gang.

Found a grand total of one issue, which was a false positive as it's in test code:

![image](https://github.com/fastify/fastify/assets/43814140/a76d7d91-cd41-4119-950c-fd9f1cfd8779)

**Comment by Uzlopak** [comment]
How did you trigger it?

**Comment by Uzlopak** [comment]
Ok found it. Well tbh. CodeQL is now set up. Is it enough to get a pass?

**Comment by ljharb** [comment]
I'd say so, I'll update the results :-)

**Comment by ljharb** [comment]
passing is now 100%; silver is now 85%.

I'll make a PR to add the passing badge, and I'll keep poking on the silver requirements - I suspect that fastify can qualify without making additional changes but I need to clarify the requirements.

**Comment by ljharb** [comment]
Filed https://github.com/fastify/fastify/pull/4926

**Comment by jsumners** [comment]
> Enabled and found a grand total of one issue, which was a false positive as it's in test code:

<details><summary>👀</summary>
<p>

![puppet-awkward](https://github.com/fastify/fastify/assets/321201/12ddc062-1896-4ff1-9a81-c14b49ab50f5)


</p>
</details>

**Comment by Uzlopak** [comment]
I ran codeql about 9 months ago on fastify

![image](https://github.com/fastify/fastify/assets/5059100/9d892938-26b4-4ba0-b366-cf1acf3c3bfc)

Also I think I checked fastify with the predecessor of codeql, lgtm.com, before and there were as far as i remember not significant issues.

**Comment by Fdawgs** [comment]
> I'm happy to set up CodeQL in GitHub Actions and do the grunt work of flushing out the false positives for this btw gang.

I've flushed out all the false positives in the other Fastify repos.

There's a handful in [fastify-static](https://github.com/fastify/fastify-static) and [fastify-swagger](https://github.com/fastify/fastify-swagger) that i'm unsure of so have not dismissed.
Would someone else be able to give them a poke please?

**Comment by Eomm** [comment]
Where should we look at?
Can't understand this output tbh https://github.com/fastify/fastify-static/actions/runs/5719373683/job/15497124421

Sometimes things are too fast to be up to date

**Comment by Uzlopak** [comment]
In the Security Tab under Code Scannning

https://github.com/fastify/fastify-static/security/code-scanning
